vnext/JSI/Shared/ChakraRuntime.h

```diff
@@ -4,53 +4,69 @@
 #pragma once
 
 #include "ChakraObjectRef.h"
-#include "ChakraRuntimeArgs.h"
 
 #include "jsi/jsi.h"
 
-#ifdef CHAKRACORE
-#include "ChakraCore.h"
-#include "ChakraCoreDebugger.h"
-#else
-#ifndef USE_EDGEMODE_JSRT
-#define USE_EDGEMODE_JSRT
-#endif
-#include "jsrt.h"
-#endif
-
 #include <memory>
-#include <mutex>
-#include <sstream>
-
-#if !defined(CHAKRACORE)
-class DebugProtocolHandler {};
-class DebugService {};
-#endif
+#include <string>
+#include <type_traits>
+#include <utility>
+#include <vector>
 
 namespace Microsoft::JSI {
 
+struct ChakraRuntimeArgs;
+
+class ChakraPreparedJavaScript : public facebook::jsi::PreparedJavaScript {
+ public:
+  ChakraPreparedJavaScript(
+      std::string &&sourceUrl,
+      const std::shared_ptr<const facebook::jsi::Buffer> &sourceCode,
+      const std::shared_ptr<const facebook::jsi::Buffer> &bytecode)
+      : m_sourceUrl{std::move(sourceUrl)}, m_sourceCode{sourceCode}, m_bytecode{bytecode} {}
+
+  inline const std::string &SourceUrl() const {
+    return m_sourceUrl;
+  }
+
+  inline const std::shared_ptr<const facebook::jsi::Buffer> &SourceCode() const {
+    return m_sourceCode;
+  }
+
+  inline const std::shared_ptr<const facebook::jsi::Buffer> &Bytecode() const {
+    return m_bytecode;
+  }
+
+ private:
+  std::string m_sourceUrl;
```

Comment:
>m_sourceUrl [](start = 14, length = 11)

nit: If we're going to offer these with direct access, should we just make them public and const?

Reply:
That makes sense. Will make the appropriate change.

---
In reply to: [345453590](https://github.com/microsoft/react-native-windows/pull/3650#discussion_r345453590) [](ancestors = 345453590)

Reply:
providing getters is better

---
In reply to: [345474793](https://github.com/microsoft/react-native-windows/pull/3650#discussion_r345474793) [](ancestors = 345474793,345453590)